bindings/tests/testenv/testenv.go

```diff
@@ -192,7 +194,9 @@ func (te *TestEnvironment) cleanup() {
 }
 
 func createClient() sui.ISuiAPI {
-	return sui.NewSuiClient(fmt.Sprintf("http://localhost:%d", instance.rpcPort))
+	return sui.NewSuiClientWithCustomClient(fmt.Sprintf("http://localhost:%d", instance.rpcPort), &http.Client{
+		Timeout: HTTPClientTimeout,
+	})
```

Comment:
You could add an optional timeout param
```suggestion
func createClient(timeout ...time.Duration) sui.ISuiAPI {
    clientTimeout := HTTPClientTimeout
    if len(timeout) > 0 {
        clientTimeout = timeout[0]
    }
    
    return sui.NewSuiClientWithCustomClient(
        fmt.Sprintf("http://localhost:%d", instance.rpcPort), 
        &http.Client{Timeout: clientTimeout},
    )
```

Reply:
Applied